modules/stochastic_tools/include/utils/Calculators.h

```diff
@@ -43,80 +49,166 @@ class Calculator : public libMesh::ParallelObject
 {
 public:
   Calculator(const libMesh::ParallelObject & other, const std::string & name)
-    : libMesh::ParallelObject(other), _name(name)
+    : libMesh::ParallelObject(other), _name(name), _state(CalculatorState::NONE)
   {
   }
 
   virtual ~Calculator() = default;
-  virtual OutType compute(const InType &, bool) const = 0;
+  OutType compute(const InType &, bool);
+
+  void initialize();
+  void update(const typename InType::value_type &);
+  void finalize(bool);
+  OutType get() const;
+
   const std::string & name() const { return _name; }
 
+protected:
+  virtual void initializeCalculator() = 0;
+  virtual void updateCalculator(const typename InType::value_type &) = 0;
+  virtual void finalizeCalculator(bool) = 0;
```

Comment:
I think these functions should be flipped, the internal methods should have the Calculator prefix (or name them intializeInternal). The non-prefixed is what should be overridden, mainly because that is how we do it with other objects in MOOSE.

Reply:
Sorry, not sure what you mean here. The public functions do not have the suffix, BoostrapCalculator calls `initialize`, `update`, and `finalize`. The internal protected functions have the suffix and are the ones meant to be overridden. I'm not sure how to write it such that the public functions are the ones being overridden, unless every calculator has the "state" checks.

Reply:
I was trying to say the "initialize", "update", and "finalize" should be the ones meant to be overridden, to be consistent with other objects in MOOSE. Have BootstrapCalculator call the methods with the prefix instead.

Reply:
Ohh, got it. I will update right now.